nilearn/plotting/tests/test_surf_plotting.py

```diff
@@ -36,6 +36,14 @@ def test_plot_surf():
     plot_surf(mesh, bg_map=bg, colorbar=True, cbar_vmin=0,
               cbar_vmax=150, cbar_tick_format="%i")
 
+    # Plot with avg_method
+    plot_surf(mesh, surf_map=rng.standard_normal(size=mesh[0].shape[0]), avg_method='mean')
```

Comment:
I'm wondering whether we should test explicitly that the texture we obtain results from the `avg_method` provided :thinking: 
I was thinking about something along those lines:

```python
import matplotlib.pyplot as plt
import numpy as np
from nilearn.surface import load_surf_mesh
from nilearn.surface.testing_utils import generate_surf
from numpy.testing import assert_array_equal
from nilearn.plotting.surf_plotting import plot_surf

mesh = generate_surf()
rng = np.random.RandomState(42)
bg = rng.standard_normal(size=mesh[0].shape[0])
mapp = rng.standard_normal(size=mesh[0].shape[0])
mesh_ = load_surf_mesh(mesh)
coords, faces = mesh_[0], mesh_[1]

for method in ['mean', 'median', 'min', 'max']:
    display = plot_surf(mesh, surf_map=mapp, avg_method=method)
    if method == 'mean':
        agg_faces = np.mean(mapp[faces], axis=1)
    elif method == 'median':
        agg_faces = np.median(mapp[faces], axis=1)
    elif method == "min":
        agg_faces = np.min(mapp[faces], axis=1)
    elif method == "max":
        agg_faces = np.max(mapp[faces], axis=1)
    vmin = np.min(agg_faces)
    vmax = np.max(agg_faces)
    agg_faces -= vmin
    agg_faces /= (vmax - vmin)
    cmap = plt.cm.get_cmap(plt.rcParamsDefault['image.cmap'])
    assert_array_equal(cmap(agg_faces),
                       display._axstack.as_list()[0].collections[0]._facecolors)
```

WDYT @alexisthual ?

Reply:
That's a very good idea, I added your suggestion